logstash-core/lib/logstash/pipeline_action/reload.rb

```diff
@@ -66,7 +66,7 @@ def execute(agent, pipelines_registry)
         # the block must emit a success boolean value
 
         # First shutdown old pipeline
-        old_pipeline.shutdown { LogStash::ShutdownWatcher.start(old_pipeline) }
+        old_pipeline.shutdown { LogStash::ShutdownWatcher.new(old_pipeline).start }
```

Comment:
this block should no longer be needed, right? `Pipeline#shutdown` will ignore the passed block (silently, unfortunately), but has its own `wait_for_shutdown` to spawn and wait out a shutdown watcher.